router/src/dml_handlers/rpc_write/circuit_breaker.rs

```diff
@@ -638,4 +640,68 @@ mod tests {
         c.requests.observe::<(), ()>(&Err(()));
         assert!(!c.is_healthy());
     }
+
+    /// A multi-threaded fuzz test that ensures no matter what sequence of
+    /// events occur, the circuit can always be driven healthy by probing
+    /// successfully.
+    #[tokio::test]
```

Comment:
I think you need to use the threaded scheduler to actually run on multiple threads
```suggestion
    #[tokio::test(threaded_scheduler)]
```

Reply:
It makes no difference. It's spawning actual threads, not tasks - if it were the latter, then it would need a multithreaded executor :+1: